rules/rules.bzl

```diff
@@ -35,68 +35,99 @@ def _licence_check_impl(ctx):
         command = 'touch "{}"'.format(checker.path),
     )
 
+    workspace = ctx.file.workspace.path if ctx.file.workspace else ""
     script = ctx.actions.declare_file(ctx.label.name + ".bash")
     ctx.actions.expand_template(
         template = ctx.file._runner,
         output = script,
         substitutions = {
             "@@LICENCE_CHECKER@@": ctx.executable.licence_check.path,
             "@@CONFIG@@": config.path,
+            "@@WORKSPACE@@": workspace,
         },
         is_executable = True,
     )
 
-    runfiles = ctx.runfiles(files = [config, checker], transitive_files = ctx.attr.licence_check.files)
+    files = [config, checker]
+    if ctx.file.workspace:
+        files.append(ctx.file.workspace)
+
+    runfiles = ctx.runfiles(files = files, transitive_files = ctx.attr.licence_check.files)
     runfiles = runfiles.merge(
-      ctx.attr.licence_check.default_runfiles,
+        ctx.attr.licence_check.default_runfiles,
     )
 
     return DefaultInfo(
         runfiles = runfiles,
         executable = script,
     )
 
+licence_check_attrs = {
+    "config": attr.label(
+        allow_single_file = True,
+        doc = "HJSON configuration file override for the licence checker",
+    ),
+    "licence": attr.string(
+        mandatory = True,
+        doc = "Text of the licence header to use",
+    ),
+    "match_regex": attr.bool(
+        default = False,
+        doc = "Whether to use regex-matching for the licence text",
+    ),
+    "exclude_patterns": attr.string_list(
+        default = [],
+        doc = "File patterns to exclude from licence enforcement",
+    ),
+    "licence_check": attr.label(
+        default = "//licence-checker",
+        cfg = "host",
+        executable = True,
+        doc = "The licence checker executable",
+    ),
+    "workspace": attr.label(
+        allow_single_file = True,
+        doc = "Label of the WORKSPACE file",
+    ),
+    "_runner": attr.label(
+        default = "//rules:licence-checker-runner.template.sh",
+        allow_single_file = True,
+    ),
+    "_config": attr.label(
+        default = "//rules:licence-checker-config.template.hjson",
+        allow_single_file = True,
+    ),
+    "_sh_runfiles": attr.label(
```

Comment:
Are we using the `_sh_runfiles` attribute anywhere in this rule, or can it be removed?

Reply:
Apparently not.  Removed.